iceoryx_posh/test/integrationtests/test_popo_chunk_building_blocks.cpp

```diff
@@ -0,0 +1,178 @@
+{
+    uint64_t dummy{42};
+};
+
+static constexpr size_t MEMORY_SIZE = 1024 * 1024;
+uint8_t m_memory[MEMORY_SIZE];
+static constexpr uint32_t NUM_CHUNKS_IN_POOL = 500;
+static constexpr uint32_t ITERATIONS = 10000;
+static constexpr uint32_t SMALL_CHUNK = 128;
+static constexpr uint32_t MAX_NUMBER_QUEUES = 128;
+
+using ChunkDistributorData_t = iox::popo::ChunkDistributorData<MAX_NUMBER_QUEUES, iox::popo::ThreadSafePolicy>;
+using ChunkDistributor_t = iox::popo::ChunkDistributor<ChunkDistributorData_t>;
+
+// Memory objects
+Allocator m_memoryAllocator{m_memory, MEMORY_SIZE};
+MePooConfig m_mempoolConfig;
+MemoryManager m_memoryManager;
+
+class ChunkBuildingBlocks_IntegrationTest : public Test
+{
+  public:
+    ChunkBuildingBlocks_IntegrationTest()
+    {
+        m_mempoolConfig.addMemPool({SMALL_CHUNK, NUM_CHUNKS_IN_POOL});
+        m_memoryManager.configureMemoryManager(m_mempoolConfig, &m_memoryAllocator, &m_memoryAllocator);
+    }
+    virtual ~ChunkBuildingBlocks_IntegrationTest()
+    {
+        /// @note One chunk is on hold due to the fact that chunkSender and chunkDistributor hold last chunk
+        EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1));
+    }
+
+    void SetUp()
+    {
+        m_chunkSender.addQueue(&m_chunkData);
+        m_chunkDistributor.addQueue(&m_chunkReceiverData);
+    }
+    void TearDown(){};
+
+    void publish()
+    {
+        for (size_t i = 0; i < ITERATIONS; i++)
+        {
+            m_chunkSender.allocate(sizeof(DummySample))
+                .on_success([&](iox::mepoo::ChunkHeader* chunkHeader) {
+                    auto sample = chunkHeader->payload();
+                    new (sample) DummySample();
+                    static_cast<DummySample*>(sample)->dummy = i;
+                    m_chunkSender.send(chunkHeader);
+                    m_sendCounter++;
+                })
+                .on_error([]() {
+                    // Errors shall never occur
+                    FAIL();
+                });
+
+            /// Add some jitter to make thread breathe
+            std::this_thread::sleep_for(std::chrono::nanoseconds(rand() % 100));
+        }
+    }
+
+    void forward()
+    {
+        while (m_run)
+        {
+            m_popper.pop().and_then([&](SharedChunk& chunk) { m_chunkDistributor.deliverToAllStoredQueues(chunk); });
+
+            /// Add some jitter to make thread breathe
+            std::this_thread::sleep_for(std::chrono::nanoseconds(rand() % 100));
+        }
+    }
+
+    void subscribe()
+    {
+        while (m_receiveCounter < ITERATIONS)
+        {
+            m_chunkReceiver.get()
+                .on_success([&](iox::cxx::optional<const iox::mepoo::ChunkHeader*>& maybeChunkHeader) {
+                    if (maybeChunkHeader.has_value())
+                    {
+                        auto chunkHeader = maybeChunkHeader.value();
+
+                        auto dummySample = *reinterpret_cast<DummySample*>(chunkHeader->payload());
+
```

Comment:
in publish you set the dummy counter but you don't check it here. Checking the order with the dummy variable would be a reasonable x-tra check

Reply:
Added the check. Somehow we're still loosing samples, the `dummySample.dummy` is not always monotonically increasing. SoFi stresstests run fine #115, so I fear it's something else.

Reply:
We figured out that the pusher was faster than the popper which led to overflow

Reply:
Exactly, with the overflow flag we can now distinguish between an overflow error or other errors in the building blocks.